client/rpc/src/system/mod.rs

```diff
@@ -182,29 +182,47 @@ impl<B: traits::Block> System<B> {
 				system.deny_unsafe.check_if_safe()?;
 				let (tx, rx) = oneshot::channel();
 				let _ = system.send_back.unbounded_send(Request::NetworkAddReservedPeer(peer, tx));
-				rx.await.map_err(oneshot_canceled_err)
+				match rx.await {
+					Ok(Ok(())) => Ok(()),
+					Ok(Err(e)) => Err(to_call_error(e)),
+					Err(e) => Err(to_call_error(e)),
+				}
```

Comment:
Is there an `and_then` hiding in here somewhere, e.g. would this work? `rx.await.and_then(|_| Ok(())).map_err(to_call_error)`?

Reply:
I think you example doesn't do the same thing, i.e, to will convert `Ok(Ok(()))` and `Ok(Err(e))` to `Ok()` which isn't the same so I prefer to keep it verbose for now.

It's the same on substrate master